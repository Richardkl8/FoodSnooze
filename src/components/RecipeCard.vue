<template>
    <div v-if="recipe" class="recipe-card">
        <img alt="recipe-image"
             class="recipe-image"
             :src="recipe.strMealThumb">
        <div class="recipe-text center">
            <custom-h3>
                <b>{{recipe.strMeal}}</b>
            </custom-h3>
            <custom-h3 v-if="isRelatedFoods">
                {{recipe.strCategory}}
            </custom-h3>
        </div>
    </div>
</template>

<script>
    export default {
        props: {
            recipe: {
                type: Object,
            },
            isRelatedFoods: {
                type: Boolean,
                default: false,
            },
        },
    };
</script>

<style lang="scss" scoped>
    .recipe-card {
        overflow: hidden;
        max-width: 50rem;
        cursor: pointer;
        display: flex;
        flex-direction: column;

        &:hover .recipe-image {
            transform: scale(1.02);
        }

        &:hover .recipe-text > h2 {
            text-decoration: underline;

        }
    }

    .recipe-image {
        max-width: 100%;
        transition:  all .3s ease-out;
    }

    .separator {
        border: 1px solid #C3C3C3;
    }

    .recipe-text {
        margin: 2rem 0;
    }

    @media screen and (max-width: 768px) {
        .recipe-card {
            flex-direction: row;
            margin-bottom: 2rem;
        }

        .recipe-image {
            width: 50%;
            height: 50%;
            margin-right: 2rem;
        }

        .recipe-text {
            margin: 0 0;
        }
    }
</style>
