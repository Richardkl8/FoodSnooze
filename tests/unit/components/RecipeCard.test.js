import { shallowMount } from '@vue/test-utils';
import flushPromises from 'flush-promises';
import RecipeCard from '../../../src/components/RecipeCard.vue';
import stubs from '../stubs/stubs';

const recipe = {
    strMealThumb: 'image.jpg',
    strMeal: 'Tuna',
    strCategory: 'Fish',
};

describe('RecipeCard', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallowMount(RecipeCard, {
            propsData: {
                recipe,
                showCategory: true,
            },
            stubs,
        });
    });

    test('Should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Should have recipe and showCategory prop', () => {
        expect(wrapper.props('recipe')).toBe(recipe);
        expect(wrapper.props('showCategory')).toBe(true);

        const title = wrapper.find('[testId="recipe-title"]');
        const category = wrapper.find('[testId="recipe-category"]');

        expect(title.text()).toBe('Tuna');
        expect(category.text()).toBe('Fish');
    });

    test('Should not show category when \'showCategory\' prop is false', async () => {
        const category = wrapper.find('[testId="recipe-category"]');
        wrapper.setProps({ showCategory: false });
        await flushPromises();
        expect(category.exists()).toBe(false);
    });
});
