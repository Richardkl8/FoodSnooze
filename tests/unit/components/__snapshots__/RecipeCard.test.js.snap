// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeCard Should match snapshot 1`] = `
<div class="recipe-card"><img alt="recipe-image" src="image.jpg" class="recipe-image">
  <div class="recipe-text center">
    <custom-h3-stub testid="recipe-title"><b>Tuna</b></custom-h3-stub>
    <custom-h3-stub testid="recipe-category">
      Fish
    </custom-h3-stub>
  </div>
</div>
`;
